adapters/networkHandlerFactory.js

```diff
@@ -1,26 +1,38 @@
 const {
-  networkHandler: BrazeNetworkHandler
-} = require("../v0/destinations/braze/util");
-const {
-  networkHandler: MarketoNetworkHandler
-} = require("../v0/destinations/marketo/util");
-const {
-  networkHandler: GenericNetworkHandler
+  GenericNetworkHandler
 } = require("./networkhandler/genericNetworkHandler");
-const {
-  networkHandler: BqStreamNetworkHandler
-} = require("../v0/destinations/bqstream/util");
+const NetWorkHandlerClassMap = require("./networkHandlersMap");
+
+class NetworkHandlerFactory {
+  constructor() {
+    this.info = "NetworkHandlerFactory";
+    // intially map would contain only the generic handler
+    this.factoryMap = new Map();
+    this.factoryMap.set("generic", new GenericNetworkHandler());
+  }
+
+  getNetworkHandler(destination) {
+    if (this.factoryMap.has(destination)) {
+      return this.factoryMap.get(destination);
+    }
 
-const handler = {
-  generic: GenericNetworkHandler,
-  braze: BrazeNetworkHandler,
-  marketo: MarketoNetworkHandler,
-  bqstream: BqStreamNetworkHandler
-};
+    // fallback to generic if destination name
+    // not in the Class Map
+    if (!NetWorkHandlerClassMap[destination]) {
+      return this.factoryMap.get("generic");
+    }
 
-module.exports = {
-  getNetworkHandler(type) {
-    const NetworkHandler = handler[type] || handler.generic;
-    return new NetworkHandler();
+    // if not found in map, adds the networkHandler object
+    // to factoryMap dynamically and returns it
+    const Handler = NetWorkHandlerClassMap[destination];
+    this.factoryMap.set(destination, new Handler());
+    return this.factoryMap.get(destination);
   }
-};
+}
+
+// Alternate option 1: expose the class and make the member method as static
+// Alternate option 2: export the class and initialize the
+// singleton object inside versioned router
+
+// directly exporting a singleton factory object
+module.exports = new NetworkHandlerFactory();
```

Comment:
If node modules are cached we will only have 1 instance of `NetworkHadlerFactory` after initialization of the server. Need to verify by generating some member property like objectId it - and printing it in the versioned router. WDYT?

Reply:
yes that should help in verifying that only one instance is being created.